File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        prev1=2
        prev2=1
        if n<=2:
            return n
        for i in range(3,n+1):
            cur=prev1+prev2
            prev2=prev1
            prev1=cur
        return prev1

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n<=2){
            return n;
        }
        int prev=1;
        int curr=2;
        for(int i=0;i<n-2;i++){
            int nex=prev+curr;
            prev=curr;
            curr=nex;
        }
        return curr;
    }
}